src/Screens/ConsultantProfile.tsx

```diff
@@ -0,0 +1,438 @@
+    }
+  }, [user]);
+
+  const handleImageUpload = async () => {
+    const result: any = await launchImageLibrary({mediaType: 'photo'});
+    if (!result?.didCancel) {
+      let name = JSON.stringify(new Date().getTime());
+      const reference = storage().ref(`images/${name}`);
+      const task = reference.putFile(result?.assets[0]?.uri);
+      task.on('state_changed', taskSnapshot => {
+        console.log(
+          `${taskSnapshot.bytesTransferred} transferred out of ${taskSnapshot.totalBytes}`,
+        );
+      });
+      task.then(async () => {
+        await storage()
+          .ref(`images/${name}`)
+          .getDownloadURL()
+          .then(async (url: any) => {
+            setProfilePicture(url);
+          })
+          .catch(e => console.error(e));
+      });
+    }
+  };
+
+  const categoryOptions = categoriesList.map(category => ({
+    label: category.name,
+    value: category.name,
+  }));
+
+  const handleUpdate = () => {
+    const ageError = isValidNumber(age, 'age');
+    const rateError = isValidNumber(rate, 'rate');
+    const experienceError = isValidNumber(experience, 'experience');
+    const descriptionError = isValidDescription(description);
+    let categoryError = '';
+    let cityError = '';
+    let imageError = '';
+    if (!category) {
+      categoryError = 'Category is required';
+    }
+    if (!city) {
+      cityError = 'City is required';
+    }
+    if (!profilePicture) {
+      imageError = 'Profile Pic is required';
+    }
+    if (
+      ageError ||
+      rateError ||
+      experienceError ||
+      categoryError ||
+      cityError ||
+      descriptionError
+    ) {
+      setAgeError(ageError);
+      setRateError(rateError);
+      setExperienceError(experienceError);
+      setDescriptionError(descriptionError);
+      setCategoryError(categoryError);
+      setCityError(cityError);
+      setImageError(imageError);
+      return;
+    } else {
+      setLoading(true);
+      setAgeError('');
+      setRateError('');
+      setExperienceError('');
+      setDescriptionError('');
+      setCategoryError('');
+      setCityError('');
+      setImageError('');
+      const updatedObject = {
+        ...user,
+        age,
+        gender,
+        profilePicture,
+        experience,
+        rate,
+        category,
+        city,
+        description,
+        isProfileComplete: true,
+      };
+      firestore()
+        .collection('Users')
+        .doc(`${user?.id}`)
+        .update(updatedObject)
+        .then(async () => {
+          const documentSnapshotAfter = await firestore()
+            .collection('Users')
+            .doc(`${user?.id}`)
+            .get();
+          const currentUserDataAfter = documentSnapshotAfter.data();
+          dispatch(setUser(currentUserDataAfter));
+          setLoading(false);
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    }
+  };
+
+
+  const onLogout = async () => {

```

Comment:
The logout functionality is also accessible in the profile. Optimize its usage by relocating this function into utilities.